lib/sass/gem_package_importer.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Sass
+  # The built-in RubyGems package importer. This loads pkg: URLs from gems.
+  class GemPackageImporter
+    # @!visibility private
+    def find_file_url(url, _canonicalize_context)
+      return unless url.start_with?('pkg:')
+
+      library, _, path = url[4..].partition('/')
```

Comment:
Normally we would have to decode the percent-encoding in the URL first, but in here we're lucky:
- The gem name part will only have "unreserved characters" in a URL and dart will unescape them as part of normalization if they are escaped. If we get a percent encoding in name it must be an invalid gem name, thus we don't have to worry.
- The subpath part we can just pass the escaped version back as part of the file URL as is.